packages/core/src/secret/index.ts

```
@@ -303,50 +324,143 @@ async function batchGetKeys(
+  // Process all relative paths in parallel
+  const results = relPaths.map(async (relPath) => {
+    const pathComponents = relPath.split('/');
+    let currentPath = prefix;
+    let parent = cache[currentPath];
+
+    // Process path components sequentially within each path
+    for (const pathComponent of pathComponents) {
+      currentPath = `${currentPath}/${pathComponent}`;
+      if (typeof cache[currentPath] === 'undefined') {
+        const index = pathComponent.endsWith("'")
+          ? parseInt(pathComponent.slice(0, -1), 10) + 2 ** 31
```

Biome (1.9.4)

[error] Use Number.parseInt instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.parseInt instead.


(lint/style/useNumberNamespace)